### PR TITLE
Windows: add detection of VS 2017

### PR DESCRIPTION
**Pull request by rainers**
Needs https://github.com/dlang/dmd/pull/6716

See https://issues.dlang.org/show_bug.cgi?id=17280 and https://issues.dlang.org/show_bug.cgi?id=17320

### CODE REVIEW

**Comment by rainers** [comment]
Should we be more conservative and prefer one of the older versions if multiple VS are installed?

**Comment by WalterBright** [comment]
Prefer the latest.

File: windows/d2-installer.nsi [hunk 1]
```diff
@@ -351,6 +352,15 @@ SectionGroupEnd
 Function DetectVSAndSDK
     ClearErrors
 
+    ReadRegStr $0 HKLM "SOFTWARE\Microsoft\VisualStudio\SxS\VS7" "15.0"
+    StrCpy $1 "VC2017"
+    IfErrors not_vc2017
+        ${LineRead} "$0\VC\Auxiliary\Build\Microsoft.VCToolsVersion.default.txt" "1" $2
```
Comment (MartinNowak) [inline windows/d2-installer.nsi]: For reference [NSIS 2.46 Users Manual#LineRead](http://anders_k.users.sourceforge.net/public/NSIS/2.46/Manual.html#lineread).

File: windows/d2-installer.nsi [hunk 2]
```diff
@@ -351,6 +352,15 @@ SectionGroupEnd
 Function DetectVSAndSDK
     ClearErrors
 
+    ReadRegStr $0 HKLM "SOFTWARE\Microsoft\VisualStudio\SxS\VS7" "15.0"
```
Comment (kdmult) [inline windows/d2-installer.nsi]: I have installed VS 2017 Pro and there is no key HKLM\SOFTWARE\Microsoft\VisualStudio\SxS\VS7.
But there is HKLM\SOFTWARE\Wow6432Node\Microsoft\VisualStudio\SxS\VS7.

**Comment by rainers** [comment]
> But there is HKLM\SOFTWARE\Wow6432Node\Microsoft\VisualStudio\SxS\VS7.

The installer is a 32-bit process so the key is translated to the Wow6432Node by a 64-bit OS.

**Comment by wilzbach** [comment]
This should have been targeted at stable. I cherry-picked it, s.t. it's part of the 2.074.2 release ;-)

https://github.com/dlang/installer/pull/228

File: windows/d2-installer.nsi (hunk 2 above)
Comment (MartinNowak) [inline windows/d2-installer.nsi]: See answer below https://github.com/dlang/installer/pull/227#issuecomment-299717799.

**Pull request merged**